### PR TITLE
247 show contact info

### LINKED ISSUE #247: Show contacts information also between departure and arrival
Contacts information - one more time...

Contacts information should be displayed when the departure is in less than 48 hours **_OR_** **between departure and arrival** <== now it's not displayed between departure & arrival

### PR DESCRIPTION
**Pull request by radkag**
Contacts information should be displayed when the departure is in less than 48 hours OR between departure and arrival <== now it's not displayed between departure & arrival.

This PR depends on #243 , so merge that first.

Closes #247.<br/><br/><br/><url>LiveURL: https://smartfaq-247-show-contact-info.surge.sh</url>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -96,6 +96,7 @@
     "isomorphic-fetch": "2.2.1",
     "luxon": "1.2.0",
     "markdown-it": "^8.4.1",
+    "mockdate": "^2.0.2",
```
Comment (git-toni) [inline package.json]: Already pointed on another PR but just as a note to remember, this should go to  `devDeps` :slightly_smiling_face:

File: src/SingleBookingPage/BookingDetail.js [hunk 2]
```diff
@@ -54,21 +54,53 @@ class BookingDetail extends React.Component<Props> {
...
+      isFuture: timeDelta !== null && timeDelta > 0,
+      isUrgent,
+    };
+  };
+
+  render() {
+    const { booking } = this.props;
+    const departureTime = this.getDepartureByType(booking);
+    const arrivalTime = this.getArrivalByType(booking);
+    const departureTimeDelta = this.decideIfIsFutureAndUrgent(departureTime);
```
Comment (git-toni) [inline src/SingleBookingPage/BookingDetail.js]: Not life-changing, but I believe the `departureTimeDelta` variable name is confusing. What about something like `departureInfo`?

File: src/SingleBookingPage/BookingDetail.js [hunk 3]
```diff
@@ -54,21 +54,53 @@ class BookingDetail extends React.Component<Props> {
...
+    const arrivalTimeDelta = this.decideIfIsFutureAndUrgent(arrivalTime);
+    const { isUrgent, timeDelta } = departureTimeDelta;
+
+    const isBetweenDepartureAndArrival =
+      !departureTimeDelta.isFuture && arrivalTimeDelta.isFuture;
+    const showContactInfo = isUrgent || isBetweenDepartureAndArrival;
```
Comment (git-toni) [inline src/SingleBookingPage/BookingDetail.js]: Do we really need to show contact info *every time* that the flight is between departure and arrival? Even if it's with a delta of 2 weeks?

Comment (radkag) [reply src/SingleBookingPage/BookingDetail.js]: True, sounds logical. I could show the contact info only if the inbound is less then 48 hours as well.

File: src/SingleBookingPage/BookingDetail.js [hunk 4]
```diff
@@ -89,14 +89,18 @@ class BookingDetail extends React.Component<Props> {
...
-    const departureTimeDelta = this.decideIfIsFutureAndUrgent(departureTime);
-    const arrivalTimeDelta = this.decideIfIsFutureAndUrgent(arrivalTime);
-    const { isFuture, isUrgent, timeDelta } = departureTimeDelta;
+    const departureInfo = this.decideIfIsFutureAndUrgent(departureTime);
+    const arrivalInfo = this.decideIfIsFutureAndUrgent(arrivalTime);
+    const { isUrgent, timeDelta } = departureInfo;
+
+    const isBetweenDepartureAndArrival =
+      !departureInfo.isFuture && arrivalInfo.isUrgent;
+    const showContactInfo = isUrgent || isBetweenDepartureAndArrival;
```
Comment (git-toni) [inline src/SingleBookingPage/BookingDetail.js]: Considering you don't seem to use `isBetweenDepartureAndArrival` anywhere else. 
Could we refactor this logic( L96 to L98) just as? :slightly_smiling_face: 
```javascript
const showContactInfo = departureInfo.isUrgent || arrivalInfo.isUrgent;
```

Comment (radkag) [reply src/SingleBookingPage/BookingDetail.js]: What does it mean `L96 to L98`?

Comment (radkag) [reply src/SingleBookingPage/BookingDetail.js]: Oh, I see. Overkill.

**Pull request merged**